app/controllers/sessions_controller.rb

```diff
@@ -2,12 +2,15 @@ class SessionsController < ApplicationController
 
   def create
     user_info = request.env['omniauth.auth']
-    session[:github] = user_info[:uid]
-    if User.exists?(uid: user_info[:uid]) == false
-      User.create_user(user_info)
+    session[:login] = user_info[:uid]
+    if Artist.exists?(uid: user_info[:uid]) == false
+      Artist.save_user_oauth_info(user_info)
       redirect_to new_shop_path
+    # elsif user.name == ''
+    #  user = Artist.find_by(uid: user_info[:uid])
```

Comment:
perhaps these lines can be removed

Reply:
Ah shoot - yes. I need to solve for when the user has signed in with google oauth but exited the registration flow without completing. Right now it will route you to an empty profile page.